crates/rooch-store/src/transaction_store/mod.rs

```diff
@@ -35,6 +37,14 @@ pub trait TransactionStore {
         let start = cursor.unwrap_or(0);
         let end = start + limit;
 
+        let gaps = match end.checked_sub(start) {
+            None => return Err(anyhow::Error::msg("end value is overflow")),
+            Some(v) => v,
+        };
+        if gaps > MAX_RESULT_AMOUNT {
+            return Err(anyhow::Error::msg("end value is overflow"));
+        }
```

Comment:
We do not need to check it in the internal storage; check the RPC argument is enough.

Reply:
Fixed